host/vm-manager/0008-Handle-proper-allocation-of-hugepages-VFs.patch

```diff
@@ -0,0 +1,326 @@
+From f77ce8870413374d48c6e2e17c7f2b0d18ea36ba Mon Sep 17 00:00:00 2001
+From: intel <prashanth.suresh@intel.com>
+Date: Wed, 30 Aug 2023 11:32:13 +0530
+Subject: [PATCH] Handle proper allocation of hugepages, VFs
+
+- Using flocks to handle proper launch of VMs.
+
+Signed-off-by: Suresh, Prashanth <prashanth.suresh@intel.com>
+---
+ src/guest/vm_builder_qemu.cc | 147 ++++++++++++++++++++++++++++++-----
+ src/guest/vm_builder_qemu.h  |  10 +++
+ 2 files changed, 139 insertions(+), 18 deletions(-)
+
+diff --git a/src/guest/vm_builder_qemu.cc b/src/guest/vm_builder_qemu.cc
+index f1a3edc..fa9e531 100644
+--- a/src/guest/vm_builder_qemu.cc
++++ b/src/guest/vm_builder_qemu.cc
+@@ -72,6 +72,7 @@ constexpr const unsigned int kPreemptTimeout = 50000;
+ constexpr const unsigned int kExecQuantum = 25;
+ 
+ constexpr const char *kQmpPowerSocket = "/tmp/qmp-pwr-socket-";
++constexpr const char *kSetupLock = "/tmp/sriov.setup.lock";
+ 
+ const std::string connectors[8] = {
+                 "HDMI-1",
+@@ -193,15 +194,9 @@ static bool SetupHugePages(const std::string &mem_size) {
+         return false;
+     }
+ 
+-    /* Get free Huge pages */
+-    int free_hp = ReadSysFile(kSys2MFreeHugePages, std::ios_base::dec);
+-
+-    if (free_hp >= mem_mb/2)
+-        return true;
+-
+     /* Get nr hugepages */
+     int nr_hp = ReadSysFile(kSys2MNrHugePages, std::ios_base::dec);
+-    int required_hp = nr_hp - free_hp + mem_mb/2;
++    int required_hp = nr_hp + mem_mb/2;
+ 
+     /* Try to allocate required huge pages */
+     WriteSysFile(kSys2MNrHugePages, std::to_string(required_hp));
```

Comment:
After allocating, I see that vm-manager will check for the nr huge page allocation with the following:

    while (nr_hp != required_hp) {
    nr_hp = ReadSysFile(kSys2MNrHugePages, std::ios_base::dec);
    if (wait_cnt < 200) {
    usleep(10000);
    wait_cnt++;
    } else {

You may want to consider increasing the wait time for allocation.
I am using 10 x 1 second wait.... while here it is 200 x 10ms = 2 second wait